src/hotspot/cpu/aarch64/gc/z/zGlobals_aarch64.cpp

```diff
@@ -139,15 +139,15 @@
 //
 
 // Default value if probing is not implemented for a certain platform: 128TB
-#define DEFAULT_MAX_ADDRESS_BIT 47
+static const size_t DEFAULT_MAX_ADDRESS_BIT = 47;
 // Minimum value returned, if probing fails: 64GB
-#define MINIMUM_MAX_ADDRESS_BIT 36
+static const size_t MINIMUM_MAX_ADDRESS_BIT = 36;
 
 static size_t probe_valid_max_address_bit() {
 #ifdef LINUX
   size_t max_address_bit = 0;
   const size_t page_size = os::vm_page_size();
-  for (int i = DEFAULT_MAX_ADDRESS_BIT; i > MINIMUM_MAX_ADDRESS_BIT && max_address_bit == 0; --i) {
+  for (size_t i = DEFAULT_MAX_ADDRESS_BIT; i > MINIMUM_MAX_ADDRESS_BIT; --i) {
```

Comment:
Should the loop condition be MINIMUM_MAX_ADDRESS_BIT *inclusive* instead of exclusive? Seems weird that it is exclusive, when we later check if max_address_bit < MINIMUM_MAX_ADDRESS_BIT return MINIMUM_MAX_ADDRESS_BIT, which seems like exactly what will happen if the loop would have advanced to MINIMUM_MAX_ADDRESS_BIT (and then finish the loop).

Reply:
Not sure how to comment a few lines below where you conditionally return MINIMUM_MAX_ADDRESS_BIT in this new cool GUI.  Anyway, you could use MIN2 to return MIN2(MINIMUM_MAX_ADDRESS_BIT, max_address_bit) to make the intention more clear.

Reply:
We are not processing MINIMUM_MAX_ADDRESS_BIT in the loop, because this is the absolute minimum that should be returned. So, regardless of the probing result, we are going to return that value. Therefor, it is a waste of time to actually perform the probe.

The check `max_address_bit < MINIMUM_MAX_ADDRESS_BIT` is done, because after the probing (if it failed and we didn't find any valid page), right after the for loop, we do a single mmap and use the returned address to calculate the `max_address_bit`. So `max_address_bit` could then be lower than `MINIMUM_MAX_ADDRESS_BIT`.

For the return part, you are right, we could simplify it by using `MAX2`, since we are interested in the highest bit, and `MINIMUM_MAX_ADDRESS_BIT` is the absolute minimum we want to return.

I will push an update for the return part.

Reply:
I see. Incremental looks good.